test/path_tokenizer.h

```diff
@@ -0,0 +1,56 @@
+//
+// Created by wkl04 on 12-12-2019.
+//
+
+#ifndef MQTTSUBSCRIPTION_PATH_TOKENIZER_H
+#define MQTTSUBSCRIPTION_PATH_TOKENIZER_H
+
+#include <algorithm>
+
+template<typename Output>
+inline void mqtt_path_tokenizer(MQTT_NS::string_view str, Output write)
+{
+    auto first = str.begin();
+    auto last = str.end();
+
+    while (true) {
```

Comment:
Wouldn't this be more appropriately expressed as a do-while? "while(true)" makes me uncomfortable.